framework/i18n/Formatter.php

```diff
@@ -1167,6 +1289,133 @@ private function formatSizeNumber($value, $decimals, $options, $textOptions)
         return [$params, $position];
     }
 
+    protected function getBaseMeasureUnit($unit, $system)
+    {
+        switch ($unit) {
+            case 'length':
+                switch ($system) {
+                    case self::UNIT_SYSTEM_IMPERIAL:
+                        return 12; // 1 feet = 12 inches
+                    case self::UNIT_SYSTEM_METRIC:
+                        return 1000; // 1 meter = 1000 millimeters
+                }
+            case 'weight':
+                switch ($system) {
+                    case self::UNIT_SYSTEM_IMPERIAL:
+                        return 7000; // 1 pound = 7000 grains
+                    case self::UNIT_SYSTEM_METRIC:
+                        return 1000; // 1 kilogram = 1000 grams
+                }
+        }
+    }
+
+    protected function getUnitMultipliers($unit, $system)
+    {
+        switch ($unit) {
+            case 'length':
+                switch ($system) {
+                    case self::UNIT_SYSTEM_IMPERIAL:
+                        return [1, 12, 36, 792, 7920, 63360];
+                    case self::UNIT_SYSTEM_METRIC:
+                        return [1, 100, 1000, 1000000];
+                }
+
+            case 'weight':
+                switch ($system) {
+                    case self::UNIT_SYSTEM_IMPERIAL:
+                        return [1, 27.34375, 437.5, 7000, 98000, 196000, 784000, 15680000];
+                    case self::UNIT_SYSTEM_METRIC:
+                        return [1, 1000, 1000000];
+                }
+        }
+    }
+
+    protected function getMeasureUnit($unit, $system, $type, $baseUnit, $params)
```

Comment:
This method was added to group all unit translations for easier extending. Unfortunately, because of the `Yii::t()` usage, that doesn't make much sense - it can't be just an array. So all translations can be moved back to specific formatters.